src/procedures/__tests__/PushDividendPayment.ts

```diff
@@ -0,0 +1,312 @@
+import { ProcedureType, DividendType, ErrorCode } from '~/types';
+import { PolymathError } from '~/PolymathError';
+import { mockFactories } from '~/testUtils/mockFactories';
+import { Factories } from '../../Context';
+import { SecurityToken, DividendDistribution } from '~/entities';
+
+const params = {
+  symbol: 'TEST1',
+  dividendIndex: 0,
+  shareholderAddresses: ['0x01'],
+  dividendType: DividendType.Eth,
+};
+
+describe('PushDividendPayment', () => {
+  let target: PushDividendPayment;
+  let contextMock: MockManager<contextModule.Context>;
+  let wrappersMock: MockManager<wrappersModule.PolymathBase>;
+  let tokenFactoryMock: MockManager<tokenFactoryModule.MockedTokenFactoryModule>;
+  let securityTokenMock: MockManager<contractWrappersModule.SecurityToken_3_0_0>;
+  let erc20DividendsMock: MockManager<contractWrappersModule.ERC20DividendCheckpoint_3_0_0>;
+  let ethDividendsMock: MockManager<contractWrappersModule.EtherDividendCheckpoint_3_0_0>;
+  let dividendFactoryMock: MockManager<dividendFactoryModule.DividendDistributionFactory>;
+  let factoriesMockedSetup: Factories;
+
+  beforeEach(() => {
+    // Mock the context, wrappers, tokenFactory and securityToken to test PushDividendPayment
+    contextMock = ImportMock.mockClass(contextModule, 'Context');
+    wrappersMock = ImportMock.mockClass(wrappersModule, 'PolymathBase');
+    tokenFactoryMock = ImportMock.mockClass(tokenFactoryModule, 'MockedTokenFactoryModule');
+    securityTokenMock = ImportMock.mockClass(contractWrappersModule, 'SecurityToken_3_0_0');
+    erc20DividendsMock = ImportMock.mockClass(
+      contractWrappersModule,
+      'ERC20DividendCheckpoint_3_0_0'
+    );
+    ethDividendsMock = ImportMock.mockClass(
+      contractWrappersModule,
+      'EtherDividendCheckpoint_3_0_0'
+    );
+    tokenFactoryMock.mock(
+      'getSecurityTokenInstanceFromTicker',
+      securityTokenMock.getMockInstance()
+    );
+    contextMock.set('contractWrappers', wrappersMock.getMockInstance());
+    wrappersMock.set('tokenFactory', tokenFactoryMock.getMockInstance());
+    dividendFactoryMock = ImportMock.mockClass(
+      dividendFactoryModule,
+      'DividendDistributionFactory'
+    );
+    factoriesMockedSetup = mockFactories();
+    factoriesMockedSetup.dividendDistributionFactory = dividendFactoryMock.getMockInstance();
+    contextMock.set('factories', factoriesMockedSetup);
+  });
+
+  afterEach(() => {
+    restore();
+  });
+
+  describe('Types', () => {
+    test('should extend procedure and have PushDividendPayment type', async () => {
+      target = new PushDividendPayment(params, contextMock.getMockInstance());
+      expect(target instanceof Procedure).toBe(true);
+      expect(target.type).toBe(ProcedureType.PushDividendPayment);
+    });
+  });
+
+  describe('PushDividendPayment', () => {
+    test('should throw if there is no valid security token being provided', async () => {
+      // Instantiate PushDividendPayment with incorrect security symbol
```

Comment:
I'm not sure this comment reflects the line below it. You can move the target = ... instantiation in the beforeEach if you want, your choice